src/NuGet.Indexing/ResponseFormatter.cs

```diff
@@ -26,7 +26,25 @@ public static class ResponseFormatter
             NuGetVersion semVerLevel,
             Query query)
         {
-            Uri baseAddress = searcher.Manager.RegistrationBaseAddress[scheme];
+            Uri baseAddress;
+            var useSemVer2Registration = SemVerHelpers.ShouldIncludeSemVer2Results(semVerLevel);
+            RegistrationAddresses registrationAddresses = searcher.Manager.RegistrationBaseAddresses;
+
+            switch (scheme)
+            {
+                case "http":
+                    baseAddress = (useSemVer2Registration ? registrationAddresses.SemVer2HttpRegistrationAddress : registrationAddresses.HttpRegistrationAddress);
+                    break;
+                case "https":
+                    baseAddress = (useSemVer2Registration ? registrationAddresses.SemVer2HttpsRegistrationAddress : registrationAddresses.HttpsRegistrationAddress);
+                    break;
+                case "test":
+                    // this case is explicitly for test purposes. When ResponseFormatterTests calls into this, it calls with "test" scheme, which we will then return the default value.
```

Comment:
It may be clearer if we have this explicitly set `baseAddress` to `Constants.BaseUrl`.